otter/convergence/gathering.py

```diff
@@ -84,21 +83,27 @@ def group_id(s):
     return get_all_server_details(changes_since).on(servers_apply)
 
 
-def merge_servers(first, second):
+def updated_deleted_servers(old, new):
     """
-    Return servers got by merging the two sets of servers where second
-    takes precedence over first
+    Return servers with servers not found in new list marked as deleted
 
-    :param list first: List of server dicts
-    :param list second: List of server dicts
-    :return: List of merged server dicts
+    :param list old: List of old servers
+    :param list second: List of latest servers
+    :return: List of updated servers
     """
+
     def sdict(servers):
         return {s['id']: s for s in servers}
 
-    return merge(sdict(first), sdict(second)).values()
+    old = sdict(old)
+    new = sdict(new)
+    deleted_ids = set(old.keys()) - set(new.keys())
+    for s in deleted_ids:
+        old[s]["status"] = "DELETED"
```

Comment:
To explain this a little more, this is how transform works:

`structure.transform(*path*, *value-or-callable*)`

It basically walks the existing data in `structure` by following `path`. Each element in `path` can either be an index or a function of `index -> Bool`. It continues walking if the Bool is True. `*value-or-callable*` is either a literal value to set as the new value at the path, or a callable to transform the existing value.

In this case, we're using `deleted_ids.__contains__` as an index -> Bool function, to say that we should operate on all values for which the server ID is in the `deleted_ids`. The next and last part of the path is the `status` key, so the value, `DELETED` will be set there.